.github/workflows/cpp-packaging.yml

```diff
@@ -35,9 +39,26 @@ env:
   xcodeVersion: "12"
 
 jobs:
+  prepare_matrix:
+    runs-on: ubuntu-latest
+    env:
+      USE_EXPANDED_MATRIX: 0
```

Comment:
Am probably missing something here but is there a reason why this is different from the `android.yml` workflow? We didn't define a separate job evn variable there.

Reply:
The other workflows drive the expanded build matrix through the python script.  In those yml files we optionally define  ${EXPANDED_MATRIX_PARAM} as is "-e=1" for expanded runs.  Its not defaulted on purpose becuase we use it's lack of exisitence to withdraw the parameter from the python command line arguments.  For instance:

`python scripts/gha/print_matrix_configuration.py -w android -k os ${EXPANDED_MATRIX_PARAM}`

if ${EXPANDED_MATRIX_PARAM} isn' defined then it resolves to `python scripts/gha/print_matrix_configuration.py -w android -k os`.  But if it is defined to `-e=1` then the python execution looks like `python scripts/gha/print_matrix_configuration.py -w android -k os -e=1`.


However, for this workflow, we don't have an expanded build matrix so we don't yet invoke the python script.  In the future I believe that the intention is to have an expanded test matrix though, and this work simply adds an exported bool so that the other jobs may be rewritten to make their own choices on what to do for an expanded test matrix.

I signal this with either a value of `1` or a `0`.  I didn't want the two optoins to be  `1` or `non-value`, so this this env directive defaults the value to `01.  If I didn't do this then the line below `echo "::set-output name=use_expanded_matrix::${USE_EXPANDED_MATRIX}"` would have resolved to `echo "::set-output name=use_expanded_matrix::"` instead of `echo "::set-output name=use_expanded_matrix::0"`, which I think is cleaner.